sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ServiceBusProcessor.cs

```diff
@@ -595,9 +597,20 @@ public virtual async Task StopProcessingAsync(CancellationToken cancellationToke
                     RunningTaskTokenSource.Dispose();
                     RunningTaskTokenSource = null;
 
+                    // Calling SessionClosingAsync event on each receiver close
+                    foreach (var receiverMap in sessionIdsReceiverMap)
+                    {
+                        if (!receiverMap.Value.IsDisposed)
+                        {
+                            await OnSessionClosingAsync(
+                                (ServiceBusSessionReceiver)receiverMap.Value,
+                                cancellationToken)
+                                .ConfigureAwait(false);
+                        }
```

Comment:
Should we dispose the receivers here?

Reply:
I don't think we need to dispose the receivers here. I believe `ActiveReceiveTask.Dispose();` will dispose everything in line 623.

Reply:
It will dispose of the Task's internal resources, but it won't handle disposing the receivers.